docs/adr/0001-add-generic-fetcher.md

```diff
@@ -0,0 +1,104 @@
+# Add generic fetcher
+
+## Context
+
+The main motivation for this change is to cover use cases of users that need to download arbitrary files that don't fit
+within an established package ecosystem cachi2 could potentially otherwise support. The target audience is users that
+want to use cachi2 to achieve hermetic builds and want an easy way to also include these arbitrary files, that cachi2
+will account for in the SBOM it produces.
+
+## Decision
+
+A new package manager for generic artifacts must be introduced.. This package manager utilizes a custom
+lockfile that is located in the input repository. Based on that lockfile, it will download files, save them into a requested
+location, and verify checksums. Below is a more detailed overview of the implementation.
+
+### Lockfile format
+
+Cachi2 expects the lockfile to be named `artifacts.lock.yaml`.
+In order to account for possible future breaking changes, the lockfile will contain a `metadata` section with a `version`
+field that will indicate the version of the lockfile format. It will also contain a list of artifacts (files) to download,
+each of the artifacts to have a URL, list of checksums, and optionally resulting filename specified.
+
+```yaml
+metadata:
+  # uses X.Y semantic versioning
+  version: "1.0"
+artifacts:
+  - download_url: https://huggingface.co/instructlab/granite-7b-lab/resolve/main/model-00001-of-00003.safetensors?download=true
+    filename: granite-model-1.safetensors
+    checksum: sha256:d16bf783cb6670f7f692ad7d6885ab957c63cfc1b9649bc4a3ba1cfbdfd5230c
+```
+
+#### Lockfile properties
+
+Below is an explanation of individual properties of the lockfile.
+
+##### download_url (required)
+
+Specified as a string containing the download url of the artifact.
+
+##### checksum (required)
+
+Specified as string in the format of "algorithm:hash". Must be provided to ensure at least some
+degree of confidence in the identity of the artifact.
```

Comment:
Non-blocking, opinion based: I suggest dropping either "at least some degree of confidence" or the entire second sentence altogether. A checksum is a mean to validate identity of an artifact, if someone insists on arbitrary files in their build and provide a checksum then we must assume that this is exactly what they want as long as the checksum matches.

Reply:
Fixed.